src/Action/Delivery/CalculateRetailPrice.php

```diff
@@ -68,10 +116,28 @@ public function __invoke(Delivery $data, Request $request)
         $taxAmount = (int) $this->calculator->calculate($amount, $taxRate);
 
         return new RetailPrice(
+            $order->getItems(),
+            $calculationOutput,
             $amount,
             $this->currencyContext->getCurrencyCode(),
             $taxAmount,
             'included' === $request->query->get('tax', 'included')
         );
     }
 }
+
+class CalculationItem {
+    /**
+     * @param RuleResult[] $rules
```

Comment:
doesnt work directly as function annotation?

Reply:
As far as I know, php does not support typed arrays. Or am I missing something?